src/lib/fees/model.ts

```diff
@@ -12,6 +17,7 @@ type CalculateFeeBpsArgs = {
   tradeAmountUsd: BigNumber
   foxHeld: BigNumber
   thorHeld?: BigNumber
+  foxWifHatHeld?: BigNumber
```

Comment:
Blocking:

1. `foxWifHatHeld` is a `bnOrZero()` BN, so *should* always be defined. I'm assuming that the intent of the optional here was to match the intent of `thorHeld` i.e `T | undefined` above, which is actually incorrect, since it's *consistently* passed, or at least it *should* have been. Looks like the real intent here was `T | undefined`, which *is* accidentally fine given we're now passed pre-TIP-014 free trades, but was previously a bug when we had these. 

```suggestion
  foxWifHatHeld: BigNumber
```

Removing the optional however, unveiled some lil' issues, keep reading...

2. While updating both `foxWifHatHeld` and `thorHeld` to remove the optional, noticed a few type errors, notably in `src/lib/fees/model.test.ts` and `src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx` and `src/components/FeeExplainer/FeeExplainer.tsx` where we never handled `thorHeld` and are now not handling `foxWifHatHeld` either, which seems incorrect?

Suggestions:

1. For this PR, make both required to match actual intent 
2. Follow-up with removal of pre TIP-014 free trades, since the year is now 2025

Reply:
Also re: above, could be `BN | undefined` for both and not `bnorZero()`ing it out in consumers but rather `bn()`, which would make the discrimination clear between undefined (nil/None) and 0 (Some, but the number just happens to be 0)

Reply:
Agree, fixed in [a3d9766](https://github.com/shapeshift/web/pull/8663/commits/a3d9766fb8ffbb18fd5aad5c1665b80a083e62e1)

Happy to remove thorswap discount in a follow up